iron-location.html

```diff
@@ -171,36 +171,40 @@
         // one when we set this.hash. Likewise for query.
         this._dontUpdateUrl = true;
         this._hashChanged();
-        this.path = window.location.pathname;
-        this.query = window.location.search.substring(1);
+        this.path = window.decodeURIComponent(window.location.pathname);
+        this.query = window.decodeURIComponent(
+            window.location.search.substring(1));
         this._dontUpdateUrl = false;
         this._updateUrl();
       },
       _getUrl: function() {
-        var url = this.path;
-        var query = this.query;
-        if (query) {
-          url += '?' + query;
+        var partiallyEncodedPath = window.encodeURI(
+            this.path).replace(/\#/g, '%23').replace(/\?/g, '%3F');
+        var partiallyEncodedQuery = '';
+        if (this.query) {
+          partiallyEncodedQuery = '?' + window.encodeURI(
+              this.query).replace(/\#/g, '%23');
```

Comment:
Are we going for the user handling the decoding of the [Fragment Identifier](https://en.wikipedia.org/wiki/Fragment_identifier)? It seems p niche anyway.